tests/report_status_test.go

```diff
@@ -11,10 +15,78 @@ import (
 	nmstatev1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate.io/v1"
 
 	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
 	appsv1client "k8s.io/client-go/kubernetes/typed/apps/v1"
 )
 
+func IsReady(pod corev1.Pod) bool {
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == corev1.PodReady &&
+			condition.Status == corev1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func WaitPodsCleanup() {
+	wait.Poll(10, 30*time.Second, func() (bool, error) {
+		podsClient := k8sClientset.CoreV1().Pods(*nmstateNs)
+		pods, err := podsClient.List(metav1.ListOptions{})
+		if err != nil {
+			return true, fmt.Errorf("error listing pods: %v", err)
+		}
+		return len(pods.Items) == 0, nil
+	})
+}
+
+func WaitPodsReady() error {
+	return wait.Poll(10, 30*time.Second, func() (bool, error) {
+		podsClient := k8sClientset.CoreV1().Pods(*nmstateNs)
+		pods, err := podsClient.List(metav1.ListOptions{})
+		if err != nil {
+			return true, fmt.Errorf("error listing pods: %v", err)
+		}
+		if len(pods.Items) == 0 {
+			return false, nil
+		}
+		for _, pod := range pods.Items {
+			if !IsReady(pod) {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+}
+
+func GetPodsLogs() string {
+	podLogOpts := corev1.PodLogOptions{}
+	podsClient := k8sClientset.CoreV1().Pods(*nmstateNs)
+	pods, err := podsClient.List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Sprintf("error listing pods: %s\n", err)
```

Comment:
we should return string, error, so it is easy to check whether we failed or nor

Reply:
Done only the first error, the other ones just make the loop skip the failed pod and continue with others.